csrc/host_ir/jit.cpp

```diff
@@ -0,0 +1,383 @@
+  llvm::Value* ndim_arg = func->getArg(1);      // int64_t   (ndim)
+  llvm::Value* strides_arg = func->getArg(2);   // int64_t* (strides)
+  llvm::Value* strides_ndim_arg = func->getArg(3); // int64_t (strides_ndim)
+
+  // Bounds checking for ndim
+  size_t compile_time_ndim = TensorDomain::noReductions(allocate->buffer()->as<TensorView>()->getLogicalDomain()).size();
+  llvm::Value* compile_time_ndim_val =
+      llvm::ConstantInt::get(int64_type, compile_time_ndim);
+  llvm::Value* cmp =
+      builder.CreateICmpEQ(ndim_arg, compile_time_ndim_val, "ndim_check");
+  llvm::Function* parent_func = builder.GetInsertBlock()->getParent();
+  llvm::BasicBlock* then_bb =
+      llvm::BasicBlock::Create(context, "if.then", parent_func);
+  llvm::BasicBlock* else_bb =
+      llvm::BasicBlock::Create(context, "if.else", parent_func);
+  builder.CreateCondBr(cmp, then_bb, else_bb);
+  builder.SetInsertPoint(else_bb);
+  llvm::Function* trap_func =
+      llvm::Intrinsic::getDeclaration(mod, llvm::Intrinsic::trap);
+  builder.CreateCall(trap_func, {});
+  builder.CreateUnreachable();
+
+  builder.SetInsertPoint(then_bb);
+
+  // Create constants for type and device from params
+  llvm::Value* dtype_constant = llvm::ConstantInt::get(int32_type, static_cast<int32_t>(type));
+  llvm::Value* device_index_constant = llvm::ConstantInt::get(int64_type, params.getDeviceId());
+
+  // Get the at::native::empty_strided_cuda function pointer (registered in the JIT)
+  llvm::Function* at_empty_strided_cuda_func = mod->getFunction("at::native::empty_strided_cuda");
+  if (at_empty_strided_cuda_func == nullptr) {
+    llvm::FunctionType* at_empty_strided_cuda_func_type = llvm::FunctionType::get(
+        void_ptr_type, {int64_ptr_type, int64_type, int64_ptr_type, int64_type, int32_type, int64_type}, false);
+    at_empty_strided_cuda_func = llvm::Function::Create(
+        at_empty_strided_cuda_func_type, llvm::Function::ExternalLinkage, "at::native::empty_strided_cuda", mod);
+  }
+
+  // Call at::native::empty_strided_cuda with constants
+  llvm::Value* result =
+      builder.CreateCall(at_empty_strided_cuda_func, {sizes_arg, ndim_arg, strides_arg, strides_ndim_arg, dtype_constant, device_index_constant});
+
+  // Return the result
+  builder.CreateRet(result);
+
+  // Verify the module
+  std::string error;
+  llvm::raw_string_ostream error_stream(error);
+  NVF_ERROR(!llvm::verifyModule(*mod, &error_stream), "LLVM module verification failed: " + error);
+
+  const bool debug_print = isDebugDumpEnabled(DebugDumpOption::HostIrJit);
+  if (debug_print) {
+    // Print the LLVM IR module
+    llvm::outs() << "=== LLVM IR ===\n";
+    mod->print(llvm::outs(), nullptr);
+  }
+}
+
+void compile(const hir::HostIrContainer* container, llvm::orc::LLJIT* jit, std::unordered_map<const kir::Allocate*, allocate_fn>& allocate_funcs_, const HostIrJitParams& params) {
+  if (allocate_funcs_.size() > 0) {
+    return;
```

Comment:
Need some comment on this.

If I'm taking a wild guess, should we assert here instead?

Reply:
It's probably needed, if we are doing inplace update.

Reply:
If we do have an inplace kernel, you wouldn't have any `allocate_funcs_` neither... I'm not sure if it would avoid double compilation.

Reply:
anyway, I don't think this logic does anything meaningful.
Let's change it into an assert instead.